lib/ReactViews/Map/Panels/SharePanel/SharePanel.tsx

```diff
@@ -50,522 +32,89 @@ interface PropTypes extends WithTranslation {
-        >
-          <div className={Styles.btnWrapper}>
-            <button
-              type="button"
-              onClick={this.toggleAdvancedOptions}
-              className={Styles.btnAdvanced}
-            >
-              <span>{t("share.btnAdvanced")}</span>
-              {this.advancedIsOpen() ? (
-                <Icon glyph={Icon.GLYPHS.opened} />
-              ) : (
-                <Icon glyph={Icon.GLYPHS.closed} />
-              )}
-            </button>
-          </div>
-          {this.advancedIsOpen() && (
-            <>
-              <Box className={DropdownStyles.section}>
-                {this.isUrlShortenable() && (
-                  <div className={Styles.advancedOption}>
-                    <Checkbox
-                      textProps={{ small: true }}
-                      id="shortenUrl"
-                      isChecked={(this.shouldShorten() as boolean) ?? false}
-                      onChange={this.onShortenClicked}
-                      className={Styles.checkbox}
-                    >
-                      <TextSpan>{t("share.shortenUsingService")}</TextSpan>
-                    </Checkbox>
-                  </div>
-                )}
-                <div className={Styles.advancedOption}>
-                  <Checkbox
-                    textProps={{ small: true }}
-                    id="includeStory"
-                    title="Include Story in Share"
-                    isChecked={this.state.includeStoryInShare}
-                    onChange={this.onIncludeStoryInShareClicked}
-                    className={Styles.checkbox}
-                  >
-                    <TextSpan>{t("includeStory.message")}</TextSpan>
-                  </Checkbox>
-                </div>
-              </Box>
-              <hr className={Styles.thinLineDivider} />
-              <div className={DropdownStyles.section}>
-                <TextSpan className={Styles.paragraph}>
-                  {t("share.embedTitle")}
-                </TextSpan>
-                <Input
-                  large
-                  dark
-                  type="text"
-                  readOnly
-                  placeholder={this.state.placeholder}
-                  value={iframeCode}
-                  onClick={e => {
-                    const target = e.target as HTMLInputElement;
-                    return target.select();
-                  }}
-                />
-              </div>
-            </>
-          )}
-        </div>
-      </div>
-    );
-  }
-
-  openWithUserClick() {
-    if (this.props.onUserClick) {
-      this.props.onUserClick();
+        </Box>
+      );
+    } else {
+      return (
+        <SharePanelContent
+          terria={terria}
+          viewState={viewState}
+          closePanel={this.closePanel}
+        />
+      );
     }
-    this.changeOpenState(true);
-    this.renderContentForStoryShare();
   }
 
   render() {
```

Comment:
I can see logically whats happening, but SharePanel is still a bit confusing with three different paths in `renderContent()` and only two different paths in `render()`.

I kind of think we should focus it more explicitly around the three modes `catalogShare`, `storyShare`, and `printEmbedShare` so that it is clearly structured for maintainability moving forward... What do you think?